specs/proposals/blob-signing.md

```diff
@@ -0,0 +1,282 @@
+- Detailed command usage, which will be covered in the individual command specifications at [Notation CLI Docs](https://github.com/notaryproject/notation/tree/v1.3.0/specs/commandline).
+- Distribution of certificates for verification across different scenarios.
+- Compatibility of the payload format. While some users are utilizing Notary Project-compliant signature envelopes, the payload format for blobs differs from the [Notary Project specification](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#blob-payload). 
+- CI/CD experience. This document specifically focuses on using Notation CLI commands.
+
+In general, all blob-related commands are grouped under the `notation blob` command for easy discovery and to avoid mixing them with OCI-based artifact commands.
+
+### Blob Sign and verify with file-based distribution
+
+For file-based distribution, such as SBOMs or release artifacts shared via a website or file transfer system, the process of signing and verifying blobs follows a similar flow. The steps below outline the user experience for signing and verifying a file, using an SBOM file named `sbom.json` as an example.
+
+**Prerequisites:**
+- SBOMs, release artifacts or other files are created and ready for publishing
+
+**Steps:**
+
+1. Sign a file on filesystem with a specified signature envelope:
+
+    ```shell
+    notation blob sign --signature-format cose sbom.json
+    ```
+
+    This generates a signature named `sbom.json.cose.sig` in the same directory as `sbom.json`. The signature file name follows the [Notary Project specification](https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#blob-signatures).
+
+2. Publish the file and corresponding signature using any file transfering mechanism. Both the file and its signature can be packaged into one file (e.g., a tarball) for transferring.
+
+3. Download or fetch both the file and signature.
+
+4. Verify the file against the signature:
+
+   - Add the root CA certificate and set up the trust store (The same experience for verifying container images):
+
+        ```shell
+        notation cert add --type ca --store myCACerts root.crt
+        ```
+
+     Confirm the certificate is added.
+
+        ```shell
+        notation cert ls
+        ```
+
+        > NOTE
+        >
+        > Learn more options using `notation cert ls --help`.
+    - Set up trust policy for blobs by adding a new command `notation blob policy init`. This command streamlines the process, eliminating the need for users to consult documentation for the correct trust policy format and preventing the accidental use of policies intended for other verification purposes.
+
+        ```shell
+        notation blob policy init --name "myBlobPolicy" --trust-store "ca:myCACerts" --trust-identity "x509.subject:C=US,ST=WA,O=wabbit-network.io"
```

Comment:
I think the verification level is `strict` and GlobalPolicy is true when using blob policy init command, right?

Reply:
Yes, other parameters will use the default values. Users can update them later according to the update guideline.

Reply:
How is the blob policy different from the standard Notation policy? Why do we need to have commands in `notation blob policy` and `notation policy`?

Reply:
@toddysm It is according to the spec https://github.com/notaryproject/specifications/blob/main/specs/trust-store-trust-policy.md#trust-policy-properties, which has been discussed and aligned within the community when AWS folks worked on this feature. Blob will use a slight different trust policy format in a different trust policy config file. So to make the experience simple for users, the new command `init` will help users to intialize a trust policy easily with default secure settings. Later on, it can be expanded to `notation policy init` command for OCI artifact, so that users do not need to do any manual edit for default config.

Reply:
See the updated comment at https://github.com/notaryproject/notation/pull/1180#discussion_r1966433336

Reply:
Aligend after offline discussion. We will follow the existing [blob CLI spec](https://github.com/notaryproject/notation/blob/main/specs/commandline/blob.md) for implementing the commands.